commonknowledge/meep-intelligence-hub
# Constituency electoral / member data popup

This PR begins the long road to political context UI.

Continues on from #31 

## Description

- List of top constituencies by member data vs total, electorate and majority
- Click a constituency (via the map or the Top Constituencies panel) to bring up a data panel, with data including:
  - current MP
  - GE2019 results or by election results, whichever more recent
  - member insights for this constituency per layer
- Member data is now differently shaded in the UI, map and in popup
- A few bug fixes (for modals not closing, map not shading properly, etc.)
- Some UX improvements: how panels look, and how they are opened and closed, UI responsiveness when clicking on the map

## Motivation and Context

The logic of this feature is to highlight the potential power of your membership to influence a local election, through sheer coalitional voting power.

We refer to this internally as the 'deliver the base' strategy, and @conatus has been studying its application in US contexts.

## How Can It Be Tested?

1. Prep the server
   1. Download the Houses of Parliament election data spreadsheet to `/data/`: https://researchbriefings.files.parliament.uk/documents/CBP-8749/HoC-GE2019-results-by-constituency.csv
   2. Rename the file to `2019_general_election.csv`
   3. Run `python manage.py import_regions` to ingest geography
   3. Run `python manage.py import_wards` to ingest geography
   3. Run `python manage.py import_mps` to ingest photography
   4. Run `python manage.py import_last_election_data` to ingest the CSV and subsequent byelections via API
2. Open a report with a member list and you should get a constituencies panel.
3. Click on a constituency on the map and the constituency electoral stats page should show up.

## How Will This Be Deployed?

1. Deploy code
3. Configure media storage on an S3 bucket (for the MP photography)
2. Run `python manage.py import_mps` to ingest photography
5. Upload the GE2019 election CSV to the `/data/` directory: https://researchbriefings.files.parliament.uk/documents/CBP-8749/HoC-GE2019-results-by-constituency.csv
6. Run `python manage.py import_last_election_data` to ingest the CSV and subsequent byelections via API

## Screenshots (if appropriate):

<img width="100%" src="https://i.imgur.com/GD00PZS.png" />

<img width="398" alt="Screenshot 2024-03-23 at 14 45 21" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/e20bb02e-2572-48aa-a8b7-d06e416ee8a6">

<img width="359" alt="Screenshot 2024-03-23 at 14 45 42" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/a923d96d-434a-40dc-af63-ff17f58cd461">

<img width="203" alt="Screenshot 2024-03-23 at 21 05 14" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/7a4bbc64-e6ab-4d64-9ca9-3aa2ad71fd07">

<img width="373" alt="Screenshot 2024-03-23 at 21 05 06" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/4c318130-2aee-41bc-b44e-0c8e2cd10c97">

<img width="276" alt="Screenshot 2024-03-23 at 14 46 50" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/2c499646-5b06-476a-9a98-38422ed02db6">

<img width="546" alt="Screenshot 2024-03-23 at 14 50 37" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/95f1f3db-6c4d-4347-a585-66ce4c5dcd60">

## Types of changes


## Checklist:
